### PR TITLE
Junits for MaconomyRestClient

### PR DESCRIPTION
**Pull request by akshay-yash**
Junits for MaconomyRestClient

### CODE REVIEW

File: src/test/java/com/imagination/cbs/util/MaconomyRestClientTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ 	@Test
+ 	public void shouldCallRestServiceForGet() {
+ 		
+ 		when(restTemplate.exchange(eq("https:baseurl/maconomy/rest"), eq(HttpMethod.GET), any(), eq(JsonNode.class))).thenReturn(new ResponseEntity<>(HttpStatus.OK));
+ 		
+ 		ResponseEntity<JsonNode> actual = maconomyRestClient.callRestServiceForGet("https:baseurl/maconomy/rest", "username", "password");
+ 		
+ 		verify(restTemplate).exchange(eq("https:baseurl/maconomy/rest"), eq(HttpMethod.GET), any(), eq(JsonNode.class));
+
+ 		assertNotNull(actual);
```
Comment (S-U-N-I-L) [inline src/test/java/com/imagination/cbs/util/MaconomyRestClientTest.java]: can we remove this? i guess not required.

File: src/test/java/com/imagination/cbs/util/MaconomyRestClientTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	
+ 	@Mock
+    private RestTemplate restTemplate;
+
+ 	@Test
+ 	public void shouldCallRestServiceForGet() {
+ 		
+ 		when(restTemplate.exchange(eq("https:baseurl/maconomy/rest"), eq(HttpMethod.GET), any(), eq(JsonNode.class))).thenReturn(new ResponseEntity<>(HttpStatus.OK));
+ 		
+ 		ResponseEntity<JsonNode> actual = maconomyRestClient.callRestServiceForGet("https:baseurl/maconomy/rest", "username", "password");
```
Comment (S-U-N-I-L) [inline src/test/java/com/imagination/cbs/util/MaconomyRestClientTest.java]: final

File: src/test/java/com/imagination/cbs/util/MaconomyRestClientTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+ 		assertNotNull(actual);
+ 		assertEquals(HttpStatus.OK, actual.getStatusCode());
+ 	}
+
+ 	@Test(expected = ResourceNotFoundException.class)
+ 	public void shouldThrowResourceNotFoundExceptionWhenRuntimeExceptionOccured() {
+
+ 		when(restTemplate.exchange(eq("https:baseurl/maconomy/rest"), eq(HttpMethod.GET), any(), eq(JsonNode.class))).thenThrow(RuntimeException.class);
+
```
Comment (S-U-N-I-L) [inline src/test/java/com/imagination/cbs/util/MaconomyRestClientTest.java]: try to avoid blank lines until necessary.

**Review by S-U-N-I-L — COMMENTED** [review]
see if there are any negative scenarios you can test for this class

File: src/test/java/com/imagination/cbs/util/MaconomyRestClientTest.java (hunk 2 above)
Comment (akshay-yash) [reply src/test/java/com/imagination/cbs/util/MaconomyRestClientTest.java]: done as per suggestion

File: src/test/java/com/imagination/cbs/util/MaconomyRestClientTest.java (hunk 1 above)
Comment (akshay-yash) [reply src/test/java/com/imagination/cbs/util/MaconomyRestClientTest.java]: done as per suggestion

File: src/test/java/com/imagination/cbs/util/MaconomyRestClientTest.java (hunk 3 above)
Comment (akshay-yash) [reply src/test/java/com/imagination/cbs/util/MaconomyRestClientTest.java]: done as per suggestion

**Pull request merged**